journal/202308.md

```diff
@@ -0,0 +1,869 @@
+- Require VSP clients to [wait for the VSP to confirm fee payments](https://github.com/decred/dcrwallet/commit/7c551ce6b95456b6ec1a174b5529641dd133230a) before considering a VSP ticket fully purchased. This eliminates multiple problematic edge cases caused by users broadcasting their own fee transactions, such as paying too low of a fee or sending payment to the wrong address. It is a preventative fix for extremely unlikely cases which have not been seen in the wild.
+
+New changes merged in `master` and backported for the next v1.8.x release:
+
+- Fixed a bug where the VSP client was not updating ticket's [live and expiry block heights](https://github.com/decred/dcrwallet/pull/2270) after it was mined. The impact of this bug was an unreasonable delay before the ticket's VSP fee was considered confirmed. There was no risk of tickets not being added to wallets or of votes being missed. Backported change is [here](https://github.com/decred/dcrwallet/commit/1319da0b05d4a9715b045e847ff58b442f407f39).
+  - { _Optionally comment on (a) how often it occurred for end users (always or in some % of cases), (b) was it purely a client-side bug or was incorrect data sent to the VSP?, (c) were wrong heights fixed automatically after some time/restart/rescan or did they persist forever?_ }
+
+User-facing changes merged in `master` towards future releases:
+
+- Added [logging of transaction changes the wallet notices](https://github.com/decred/dcrwallet/pull/2275) during rescans. This is off during the initial rescan (during reseed or after restarting dcrwallet) because it would be too noisy. Any other rescans are typically done to fix a problem, and dcrwallet will log when it finds new wallet transactions inside an old block, or wallet transactions which were previously unmined, but are now mined. Rescans aren't fun for users, and rescans are really just a bandaid to sync wallets that got out of sync. Having the hashes of missed transactions will be helpful to debug why a wallet had a problem syncing them, and will hopefully result in some bugfixes that reduce user's need to rescan.
+- Added a [cancellation check](https://github.com/decred/dcrwallet/pull/2276) to non-mixed ticket purchases before publishing the purchase. Mixed ticket purchases already check for cancellation, so this may solve an edge-case bug.
+- Fixed some error-handling logic and added new error logs [if processing VSP tickets fails](https://github.com/decred/dcrwallet/pull/2259).
+  - { _what bugs were experienced by end users when `NotExist` errors were handled incorrectly?_ }
```

Comment:
@jholdstock what about this one? 2259 called it two bug fixes. The first commit adds missed logging if adding tickets to the VSP client failed. I could not understand what end-user bug was fixed by the second commit that changed error handling in `ProcessManagedTickets`.

Reply:
The first commit of 2250 was adding adding some missing logging to make users aware that an issue exists and may need investigating.

The second commit was fixing a bit of code which was technically incorrect but I don't think caused any issues in the real world. If when adding multiple tickets to the VSP client one of them were to fail, the code would error out immediately and not attempt to add the others. With this change, all tickets are attempted, and any errors which occur are collected and returned as one at the end.

Reply:
Thanks, added in 34f45a26.